lib/internal/modules/cjs/loader.js

```diff
@@ -553,18 +559,42 @@ Module._findPath = function(request, paths, isMain) {
     return entry;
 
   let exts;
-  let trailingSlash = request.length > 0 &&
-    StringPrototypeCharCodeAt(request, request.length - 1) ===
-    CHAR_FORWARD_SLASH;
-  if (!trailingSlash) {
-    trailingSlash = RegExpPrototypeExec(trailingSlashRegex, request) !== null;
+  const trailingSlash = request.length > 0 &&
```

Comment:
I believe storing request.length in a separate variable will result in faster code execution since the next 30~40 lines call `request.length`

Reply:
Historically caching length lookups caused a huge perf hit, since it defeats a JIT optimization - not sure if that’s still true, but i doubt caching the length can *beat* the JIT.

Reply:
I'd be up for landing this as is, and open a new PR to run benchmarks and see if that changes something.